### PR TITLE
Add rate limiter

### LINKED ISSUE #149: Rate limiting

### PR DESCRIPTION
**Pull request by wblankenship**
Right now it is rather aggressive. It will require tweaking once we get better insight into typical usage.

Closes #149

### CODE REVIEW

File: src/backend/server.js [hunk 1]
```diff
@@ -75,11 +77,27 @@ passport.deserializeUser(wrap(async (id, done) => {
...
+app.enable('trust proxy'); // don't rate limit heroku
+
 const sessionStore = new SequelizeStore({
   db: models.sequelize,
   table: 'Session'
 })
 
+// List the routes that need to be rate limited
+var rateLimitRoutes = [
+  "/graphql",
```
Comment (saikat) [inline src/backend/server.js]: What's the default limit?  Any fetch of the main site will hit /graphql.

**Comment by saikat** [comment]
Did you test it somehow?  e.g. a quick script to just hammer /graphql?

**Comment by wblankenship** [comment]
Yup. I tested it by refreshing on pages :-) It does rate limit.

**Comment by saikat** [comment]
Ok nice. How aggressive is it?

**Comment by wblankenship** [comment]
It is per-second. So if you hold down refresh for a little less than a second you get banned for a few seconds. The ban is a function of the number of requests you make within your ban period. So once you are banned, the duration of your ban gets longer every additional request you make.

**Comment by wblankenship** [comment]
I think for now, this should be good. But I don't fully understand how the graphql (and similar) endpoints work.

We may find that we need to create different rate limiters for different endpoints.

**Pull request merged**